pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java

```diff
@@ -102,6 +102,10 @@ public static boolean isValid(String topic) {
         }
     }
 
+    public static String getPartitionPattern(String topic) {
+        return "^" + get(topic).getPartitionedTopicName().toString() + "-partition-[0-9]+$";
```

Comment:
Use Pattern.quote to escape possible pattern characters in the topic name

Reply:
Good suggestion, fixed